components/brave_wallet_ui/components/market-datatable/index.tsx

```diff
@@ -0,0 +1,164 @@
+    priceChange24h,
+    priceChangePercentage24h,
+    marketCap,
+    marketCapRank,
+    totalVolume
+  } = coinMarkDataItem
+
+  const formattedPrice = formatFiatAmountWithCommasAndDecimals(currentPrice.toString(), 'USD')
+  const formattedPercentageChange = formatPricePercentageChange(priceChangePercentage24h, 2, true)
+  const formattedMarketCap = formatPriceWithAbbreviation(marketCap.toString(), 'USD', 1)
+  const formattedVolume = formatPriceWithAbbreviation(totalVolume.toString(), 'USD', 1)
+  const isDown = priceChange24h < 0
+
+  const cellsContent: React.ReactNode[] = [
+    <AssetsColumnWrapper>
+      {/* Hidden until wishlist feature is available on the backend */}
+      {/* <AssetsColumnItemSpacer>
+          <AssetWishlistStar active={true} />
+        </AssetsColumnItemSpacer> */}
+      <AssetsColumnItemSpacer>
+        <TextWrapper alignment="center">{marketCapRank}</TextWrapper>
+      </AssetsColumnItemSpacer>
+      <AssetNameAndIcon
+        assetName={name}
+        symbol={symbol}
+        assetLogo={image}
+      />
+    </AssetsColumnWrapper>,
+
+    // Price Column
+    <TextWrapper alignment="right">{formattedPrice}</TextWrapper>,
+
+    // Price Change Column
+    <TextWrapper alignment="right">
+      <AssetPriceChange
+        isDown={isDown}
+        priceChangePercentage={formattedPercentageChange}
+      />
+    </TextWrapper>,
+
+    // Market Cap Column
+    <TextWrapper alignment="right">{formattedMarketCap}</TextWrapper>,
+
+    // Volume Column
+    <TextWrapper alignment="right">{formattedVolume}</TextWrapper>
+
+    // Line Chart Column
+    // Commented out because priceHistory data is yet to be
+    // available from the backend
+    // <LineChartWrapper>
+    //   <LineChart
+    //     priceData={priceHistory}
+    //     isLoading={false}
+    //     isDisabled={false}
+    //     isDown={isDown}
+    //     isAsset={true}
+    //     onUpdateBalance={() => {}}
+    //     showPulsatingDot={false}
+    //     showTooltip={false}
+    //     customStyle={{
+    //       height: '20px',
+    //       width: '100%',
+    //       marginBottom: '0px'
+    //     }}
+    //   />
+    // </LineChartWrapper>
+  ]
+
+  const cells: Cell[] = cellsContent.map(cellContent => {
+    return {
+      content: cellContent
+    }
+  })
+
+  return cells
+}
+
+export const MarketDataTable = (props: Props) => {
+  const { headers, coinMarketData, moreDataAvailable, showEmptyState, onShowMoreCoins, onSort } = props
+  const ref: any = React.useRef<HTMLDivElement>()
+  const onScreen = useOnScreen<HTMLDivElement>(ref, onScreenOptions)
+
+  React.useEffect(() => {
+    if (onScreen && moreDataAvailable) {
+      onShowMoreCoins()
+    }
+  }, [onScreen, moreDataAvailable])
+
+  const rows: Row[] = React.useMemo(() => {
+    const cells = coinMarketData.map((coinMarketItem: BraveWallet.CoinMarket) => {
+      return renderCells(coinMarketItem)
+    })
+
+    return cells.map(cell => { return { content: cell } })
```

Comment:
this is causing extra unneeded iterations, combine with the first mapping as recommended above.